Makefile

```diff
@@ -3,59 +3,66 @@ LOCALEPREFIX ?= $(PREFIX)/share/locale
 MANPREFIX    ?= $(PREFIX)/share/man
 PANDOC       ?= $(shell which pandoc)
 XDGPREFIX    ?= /etc/xdg
+SCRIPTS      ?= $(shell find ./bin -type f -executable -printf "%f\n")
+MANPAGES     ?= $(shell find ./doc -type f -name "*.md")
 
 .PHONY: setup
 setup:
 	command -v cram   || aurget cram
 	command -v vbump  || aurget vbump-git
 	command -v pandoc || stack install pandoc
 
-locale/downgrade.pot: bin/downgrade
+locale/%.pot: bin/%
 	xgettext \
 		--from-code=utf-8 -L shell \
-		--package-name=downgrade \
+		--package-name=$* \
 		--copyright-holder=$(AUTHOR) \
 		-o $@ $<
-	find ./locale/*po -exec msgmerge --update {} $@ \;
+	find ./locale/$* -name "*.po" -exec msgmerge --update {} $@ \;
 
 .PHONY: locale
-locale: locale/downgrade.pot
+locale: $(SCRIPTS:%=locale/%.pot)
 
-doc/downgrade.8: doc/downgrade.8.md
+doc/%: doc/%.md
 	$(PANDOC) --standalone --to man $< -o $@
 
 .PHONY: man
-man: doc/downgrade.8
+man: $(MANPAGES:%.md=%)
 
 .PHONY: test
 test:
 	cram test
 
 .PHONY: install
 install:
-	install -Dm755 bin/downgrade $(DESTDIR)$(PREFIX)/bin/downgrade
-	install -Dm755 bin/pacignore $(DESTDIR)$(PREFIX)/bin/pacignore
 	install -Dm644 conf/downgrade.conf $(DESTDIR)$(XDGPREFIX)/downgrade/downgrade.conf
 	install -Dm644 doc/downgrade.8 $(DESTDIR)$(MANPREFIX)/man8/downgrade.8
-	install -Dm644 completion/bash $(DESTDIR)$(PREFIX)/share/bash-completion/completions/downgrade
-	install -Dm644 completion/zsh $(DESTDIR)$(PREFIX)/share/zsh/site-functions/_downgrade
-	install -Dm644 completion/fish $(DESTDIR)$(PREFIX)/share/fish/vendor_completions.d/downgrade.fish
-	for po_file in locale/*.po; do \
+	install -Dm644 doc/pacignore.8 $(DESTDIR)$(MANPREFIX)/man8/pacignore.8
+	for script in $(SCRIPTS); do \
+	  install -Dm755 bin/$$script $(DESTDIR)$(PREFIX)/bin/$$script; \
+	  install -Dm644 completion/$$script/bash $(DESTDIR)$(PREFIX)/share/bash-completion/completions/$$script; \
```

Comment:
Two questions regarding (un)installation targets:

1. Were you suggesting to replace the for-loop over `$(SCRIPTS)` with individual lines too? I don't see a major disadvantage in the for-loop here because we intend to install all executables in `bin` and each executable should have shell completions and locales associated with it. But maybe there are edge cases and then individual lines make more sense.
2. Should we quote all paths in `Makefile` targets?

Reply:
> Were you suggesting to replace the for-loop for installing executables

I wasn't, I'm fine either way on that one. The same arguments as man pages do apply: it's a small rarely-changing list, so I would probably write it out myself. However, it was dealing with the `man8` situation that made me suggest removing that loop, which doesn't apply to the executables one.

> Should we quote

I doesn't hurt, but I don't think we need to. We fully control all of the files involved here and know they will never contain spaces.

Reply:
> I wasn't, I'm fine either way on that one

Perfect, then will keep that loop

> I doesn't hurt

Quoted the paths and arguments which have shell/Makefile variables just to be on the safe side, though I agree this should never happen in normal cases